docs/HTTPSCertificates.md

```diff
@@ -0,0 +1,24 @@
+
+## Self-signed certificate
+
+This requires the client accepting un-trusted certificates, and is the default option for the clearwater-nginx package. It involves the server creating its own CSR, which it uses to sign its certificate. Currently the post-install script for clearwater-nginx handles this using the openssl library.
+
+    openssl req -nodes -sha256 -newkey rsa:2048 -keyout nginx.key -out nginx.csr -config nginx_openssl_config
+    openssl x509 -sha256 -req -in nginx.csr -signkey nginx.key -out nginx.crt
+
+The first command creates a private key and a CSR from a very basic default set of config. The second command creates a signed certificate using the private key and the CSR. Since this is all done at start of life, clients won't be able to verify the certificate, and so will have to accept un-trusted certificates, which is not secure.
+
+## Certificate signed by the service provider as a CA
+
+This involves generating the CSR before deploying nginx. This CSR then needs to be uploaded to all clients that need to contact the nginx server, and the nginx server itself. It can then be used to create a signed certificate on the nginx server which the clients can verify. This is the recommended option, but it does require being able to upload the certificate to your clients.
+
+The CSR can be generated using openssl, as in the first option above. The config file should have the format of [nginx_openssl_config](https://github.com/Metaswitch/clearwater-nginx/blob/master/clearwater-nginx/etc/nginx/ssl/nginx_openssl_config), with the default fields replaced. The CSR and private key can then be uploaded onto the nginx server and used to generate a signed certificate, again as in the first option above. The CSR then needs to be uploaded to your clients.
+
+There are other means of generating your own CSR. However, note that bullet 1) of section 5.3.0 of the 3GPP specification 33.222 suggests that the client should check that the FQDN in the certificate is the same as the FQDN used to set up the HTTPS connection.
+
+## Certificate signed by a well known CA
+
+This involves paying an existing trusted CA to sign your certificates.
```

Comment:
This could do with some more body, even if it's simply to say "The steps required vary from CA to CA, but in general you will need to produce a CSR (see above) which you will send, along with proof of identity to your CA who will send you back a signed certificate along with their certificate chain.".  Maybe list some common CAs?  Verisign/Thawte?